source/mqtt/Mqtt5Client.cpp

```diff
@@ -389,20 +410,59 @@ namespace Aws
                 }
 
                 toSeat = new (toSeat) Mqtt5Client(options, allocator);
-                return std::shared_ptr<Mqtt5Client>(
+                // Creation failed, make sure we release the allocated memory
+                if (!*toSeat)
+                {
+                    Crt::Delete(toSeat, allocator);
+                    return nullptr;
+                }
+
+                std::shared_ptr<Mqtt5Client> shared_client = std::shared_ptr<Mqtt5Client>(
                     toSeat, [allocator](Mqtt5Client *client) { Crt::Delete(client, allocator); });
+                shared_client->m_selfReference = shared_client;
+                return shared_client;
             }
 
             Mqtt5Client::operator bool() const noexcept { return m_client != nullptr; }
 
             int Mqtt5Client::LastError() const noexcept { return aws_last_error(); }
 
-            bool Mqtt5Client::Start() const noexcept { return aws_mqtt5_client_start(m_client) == AWS_OP_SUCCESS; }
+            bool Mqtt5Client::Start() noexcept
+            {
+                if (aws_rw_lock_try_rlock(&m_client_lock) != AWS_ERROR_SUCCESS || m_client == nullptr)
+                {
+                    AWS_LOGF_ERROR(
+                        AWS_LS_MQTT5_CLIENT,
+                        "Mqtt5Client: Invalid Client or the client is in termination. Please create a new client.");
```

Comment:
Minor: I'd either change `Invalid Client ...` to `Invalid client` or change `... the client is ...` to `... the Client is ...` for consistency. Same change will be needed for `stop()`, `publish()`, etc.